src/sentry/integrations/source_code_management/metrics.py

```diff
@@ -0,0 +1,37 @@
+from enum import Enum
+
+from attr import dataclass
+
+from sentry.integrations.base import IntegrationDomain
+from sentry.integrations.utils.metrics import EventLifecycleMetric, EventLifecycleOutcome
+
+
+class RepositoryIntegrationInteractionType(Enum):
+    """
+    A specific method used by a RepositoryIntegration (???)
+    """
+
+    GET_STACKTRACE_LINK = "GET_STACKTRACE_LINK"
+    GET_CODEOWNER_FILE = "GET_CODEOWNER_FILE"
+    CHECK_FILE = "CHECK_FILE"
+
+    def __str__(self) -> str:
+        return self.value.lower()
+
+
+@dataclass
+class RepositoryIntegrationInteractionEvent(EventLifecycleMetric):
+    """
+    An instance to be recorded of ... (???)
```

Comment:
Suggestions for this docstring are appreciated, because I'm not sure what this should be.